packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentStackTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, stackName: string, stackSdk: ISDK) : Promise<Template> {
+    // if `stackName !== stackArtifact.stackName`, then `stackArtifact` is an ancestor to a nested stack with name `stackName`.
+    debug(`Reading existing template for stack ${stackArtifact.displayName}.`);
+    const cfn = stackSdk.cloudFormation();
+    const stack = await CloudFormationStack.lookup(cfn, stackName);
+
+    return stack.template();
+  }
+
+  private async prepareSdkForLookup(stackArtifact: cxapi.CloudFormationStackArtifact): Promise<ISDK> {
+    let stackSdk: ISDK | undefined = undefined;
+    // try to assume the lookup role and fallback to the deploy role
+    try {
+      const result = await prepareSdkWithLookupRoleFor(this.sdkProvider, stackArtifact);
+      if (result.didAssumeRole) {
+        stackSdk = result.sdk;
```

Comment:
Can't we just `return result.sdk` here? And this way, we can completely get rid of the `if` below?

So, the entire method would become:

```ts
  private async prepareSdkForLookup(stackArtifact: cxapi.CloudFormationStackArtifact): Promise<ISDK> {
    // try to assume the lookup role
    try {
      const result = await prepareSdkWithLookupRoleFor(this.sdkProvider, stackArtifact);
      if (result.didAssumeRole) {
        return result.sdk;
      }
    } catch { }
    // fallback to the deploy role
    return (await this.prepareSdkFor(stackArtifact, undefined, Mode.ForReading)).stackSdk;
  }
```

Which reads much better to me.